internal/services/automation/automation_watcher.go

```diff
@@ -0,0 +1,241 @@
+package automation
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/automation/mgmt/2020-01-13-preview/automation"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type WatcherModel struct {
+	ResourceGroupName           string                 `tfschema:"resource_group_name"`
+	AutomationAccountName       string                 `tfschema:"automation_account_name"`
+	Name                        string                 `tfschema:"name"`
+	Location                    string                 `tfschema:"location"`
+	Tags                        map[string]interface{} `tfschema:"tags"`
+	Etag                        string                 `tfschema:"etag"`
+	ExecutionFrequencyInSeconds int64                  `tfschema:"execution_frequency_in_seconds"`
+	ScriptName                  string                 `tfschema:"script_name"`
+	ScriptParameters            map[string]interface{} `tfschema:"script_parameters"`
+	ScriptRunOn                 string                 `tfschema:"script_run_on"`
+	Description                 string                 `tfschema:"description"`
+	Status                      string                 `tfschema:"status"`
+}
+
+type WatcherResource struct{}
+
+var _ sdk.Resource = (*WatcherResource)(nil)
+
+func (m WatcherResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"resource_group_name": commonschema.ResourceGroupName(),
+		"automation_account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"location": commonschema.Location(),
+		"tags":     commonschema.Tags(),
+		"etag": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"execution_frequency_in_seconds": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ValidateFunc: nil,
+		},
+		"script_name": {
```

Comment:
white space between properties please
```suggestion
		"resource_group_name": commonschema.ResourceGroupName(),
		
		"automation_account_name": {
			Type:         pluginsdk.TypeString,
			Required:     true,
			ForceNew:     true,
			ValidateFunc: validation.StringIsNotEmpty,
		},
		
		"name": {
			Type:         pluginsdk.TypeString,
			Required:     true,
			ForceNew:     true,
			ValidateFunc: validation.StringIsNotEmpty,
		},
		
		"location": commonschema.Location(),
		
		"tags":     commonschema.Tags(),
		
		"etag": {
			Type:         pluginsdk.TypeString,
			Optional:     true,
			ValidateFunc: validation.StringIsNotEmpty,
		},
		
		"execution_frequency_in_seconds": {
			Type:         pluginsdk.TypeInt,
			Required:     true,
			ValidateFunc: nil,
		},
		
		"script_name": {
```